### ISSUE TITLE
Real Time clock doesn't work in retroarch on devices that have no RTC backups.

### ISSUE THREAD

**Issue by majoraswrath360** [telefang/telefang#442]
From the hakchi/retroarch group: Tried Telefang on a couple different emulators. On mgba saving and reloading put me forward by a number of hours so I assumed that saving might automatically throw you forward by a set number by hours but after repeating the process I was still on the same random time it put me forward by. The only real way I would see playing it is by using mgba and using a suspend state and then loading, which didn't work on other cores for some reason but suspends work for it on mgba. The only problem is that obviously you'd have to pick up from the exact time that you last left the game
Tl:dr the real time clock is broken so it either needs to be fixed or there needs to be a way to set it ingame.

Easiest solution?: Make a similar patch to other GBC/GBA games that need relatime clock. A way to set the time ingame through an extra option in the options menu. Or through holding a certain button combination to increase/decrease the clock time.

**Comment by andwhyisit** [comment]
The no-rtc code we wrote for this game should save whatever the time was and load it when the game loads. So if you started a new game at 12pm and saved after playing for 2 hours then the clock will start at 2pm when you load your saved game. That's why immediately loading and resaving your game did nothing.

**Comment by majoraswrath360** [comment]
That is not how a real time clock works. Or how it was intended in this game. The clock should match the real time in the world we live in.

As described there where many attempts and either the clock got out of sync or just resets to 00:00.

Again the only real solution is the same one used in the pokemon games and that is to have a way to set the time manually in the game itself.

**Comment by obskyr** [comment]
I'm afraid that if the emulator doesn't properly support RTCs, there's no way to make it match the real world! For one, an adjustment feature doesn't exist in the original game, so adding one is beyond the scope of a fan translation. Secondly, adding a menu option for it may be conducive to time travel-style cheating, and would thus change the game design significantly. Thirdly, the bug in Retroarch's mGBA support is an issue on their end.

Auspiciously enough, however, good news are on the horizon: I spoke to Endrift, the developer of mGBA, and she told me that this bug is already fixed in the Retroarch version of mGBA! The fix might not be in the current latest version, but stay updated and this issue will be resolved soon.
